src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -16,6 +16,48 @@
 MSI_LOCAL_ID = '[system]'
 
 
+def _validate_template_input(namespace):
+    if namespace.template_file and not os.path.isfile(namespace.template_file):
+        raise CLIError('Please enter a valid file path')
+
+
+def _validate_template_spec(namespace):
+    if namespace.template_spec is None:
+        if (namespace.name is None or namespace.resource_group_name is None):
+            raise CLIError('incorrect usage: Please enter '
+                           'a resource group and resource name or a resource ID for --template-spec')
+    else:
+        from msrestazure.tools import is_valid_resource_id
```

Comment:
Use "from azure.mgmt.core.tools import is_valid_resource_id" as we'll deprecate "msrestazure.tools" in the future.